core/http-auth-aws-crt/src/main/java/software/amazon/awssdk/http/auth/aws/crt/AwsCrtV4aHttpSigner.java

```diff
@@ -27,6 +30,73 @@
 @SdkPublicApi
 public interface AwsCrtV4aHttpSigner extends HttpSigner<AwsCredentialsIdentity> {
 
+    /**
+     * The AWS region name to be used for computing the signature.
+     * This property is required.
+     */
+    SignerProperty<String> REGION_NAME =
+        SignerProperty.create(String.class, "RegionName");
+
+    /**
+     * The name of the AWS service.
+     * This property is required.
+     */
+    SignerProperty<String> SERVICE_SIGNING_NAME =
+        SignerProperty.create(String.class, "ServiceSigningName");
+
+    /**
+     * A {@link Clock} to be used at the time of signing.
+     * This property defaults to the time at which signing occurs.
+     */
+    SignerProperty<Clock> SIGNING_CLOCK =
+        SignerProperty.create(Clock.class, "SigningClock");
+
+    /**
+     * The name of the header for the checksum.
+     * This property is optional.
+     */
+    SignerProperty<String> CHECKSUM_HEADER_NAME =
+        SignerProperty.create(String.class, "ChecksumHeaderName");
+
+    /**
+     * The {@link ChecksumAlgorithm} used to compute the checksum.
+     * This property is required *if* a checksum-header name is given.
+     */
+    SignerProperty<ChecksumAlgorithm> CHECKSUM_ALGORITHM =
+        SignerProperty.create(ChecksumAlgorithm.class, "ChecksumAlgorithm");
```

Comment:
I'm not keen on putting a closed enum into an interface like this. We'd like to be able to declare other algorithms in separate modules. Is there a way to make this a string or a more open enum type?

Reply:
`ChecksumAlgorithm` is meant to be open/public, but it's protected for now because it was copied over.

Reply:
Though, I don't see a way to define algorithms in other modules with the way some of the `SdkChecksum` code is written (copied/adapted).

Reply:
My current thought then is to make `ChecksumAlgorithm` an interface (that extends `SdkChecksum`, and defines a method `base64EncodedLength()`) instead of an enum, and update the current `SdkChecksum` implementations to implement `ChecksumAlgorithm` instead.

Reply:
(or preferably add a new method on `SdkChecksum` for the length without needing to add a new interface)